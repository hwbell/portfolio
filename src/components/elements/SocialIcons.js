import React from 'react';

// elements

// styling
import ReactCSSTransitionGroup from 'react-addons-css-transition-group';

// element data
const icons = [
  {
    class: 'fab fa-github blink blink-grey',
    linkTo: 'https://github.com/hwbell'
  },
  {
    class: 'fab fa-codepen blink blink-orange',
    linkTo: 'https://codepen.io/hbellatcodepen/'
  },
  {
    class: 'fab fa-linkedin blink blink-blue',
    linkTo: 'https://www.linkedin.com/in/harold-bell-291926180/'
  },
  {
    class: 'fab fa-facebook-square blink blink-dark-blue',
    linkTo: 'https://www.facebook.com/profile.php?id=100010514613107'
  },
  {
    class: 'fab fa-instagram blink blink-purple',
    linkTo: 'https://www.instagram.com/harrywbell/'
  }
]

class SocialIcons extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
    }
  }

  render() {
    return (
      <div style={styles.iconContainer} className="social-icons">
        {icons.map((icon, i) => {
          return (
            <div key={i} className="animated-icon" style={styles.animatedIcon}>
              <a
                href={icon.linkTo}
                target="_blank"
                className=""
                style={styles.link}
              >
                <i
                  className={icon.class}
                  style={styles.icon}
                ></i>
              </a>
            </div>
          )
        })}
      </div>
    );
  }
}

const styles = {
  iconContainer: {
    display: "flex",
    flexDirection: "row",
    justifyContent: "flex-end",
    // width: "80%",  
    margin: "25px auto",
  },
  animatedIcon: {
    margin: 8,
  },
  link: {
    color: "whitesmoke",
  },
  icon: {
    fontSize: 40,
    // color: 'black',
  }
}

export default SocialIcons;
