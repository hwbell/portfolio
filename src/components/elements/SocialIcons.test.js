import React from 'react';
import ReactDOM from 'react-dom';
import SocialIcons from './SocialIcons';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';

// react router context
// import { MemoryRouter } from 'react-router-dom';

describe('SocialIcons Component', () => {

  let wrapper;
  let someProps;
  beforeEach(() => {
    someProps = {

    }

    wrapper = shallow(
      // we only need the MemoryRouter if there are links to routes of the react-router(pages) in the component that we need to test
      // <MemoryRouter>
      <SocialIcons {...someProps} />
      // </MemoryRouter>
    );

  })

  // tests
  it('renders without crashing', async () => {
    wrapper.update();
  });

  it('renders correctly', async () => {
    const tree = await renderer
      .create(<SocialIcons {...someProps} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('contains the correct elements for different data', () => {
    expect(wrapper.find('.social-icons').length).toEqual(1);
    expect(wrapper.find('.animated-icon').length).toEqual(5);
  })

  // it('calls click event', () => {
  //   const FakeFun = jest.spyOn(SocialIcons.prototype, 'renderIcons');
  //   const newWrapper = shallow((<SocialIcons />));
  //   newWrapper.update();
  //   expect(FakeFun).toHaveBeenCalled();
  // });

  // it('should render all buttons with info from provided props.buttons', () => {
  //   expect(wrapper.find('Link').at(0).props().to).toBe(someProps.buttons[0].link)
  //   expect(wrapper.find('Link').at(1).props().to).toBe(someProps.buttons[1].link)
  //   expect(wrapper.find('Link').at(2).props().to).toBe(someProps.buttons[2].link)

  // })
})