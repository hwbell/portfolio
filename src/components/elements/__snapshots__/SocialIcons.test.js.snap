// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialIcons Component renders correctly 1`] = `
<div
  className="social-icons"
  style={
    Object {
      "display": "flex",
      "flexDirection": "row",
      "justifyContent": "flex-end",
      "margin": "25px auto",
    }
  }
>
  <div
    className="animated-icon"
    style={
      Object {
        "margin": 8,
      }
    }
  >
    <a
      className=""
      href="https://github.com/hwbell"
      style={
        Object {
          "color": "black",
        }
      }
      target="_blank"
    >
      <i
        className="fab fa-github blink blink-grey"
        style={
          Object {
            "fontSize": 40,
          }
        }
      />
    </a>
  </div>
  <div
    className="animated-icon"
    style={
      Object {
        "margin": 8,
      }
    }
  >
    <a
      className=""
      href="https://codepen.io/hbellatcodepen/"
      style={
        Object {
          "color": "black",
        }
      }
      target="_blank"
    >
      <i
        className="fab fa-codepen blink blink-orange"
        style={
          Object {
            "fontSize": 40,
          }
        }
      />
    </a>
  </div>
  <div
    className="animated-icon"
    style={
      Object {
        "margin": 8,
      }
    }
  >
    <a
      className=""
      href="https://www.linkedin.com/in/harold-bell-291926180/"
      style={
        Object {
          "color": "black",
        }
      }
      target="_blank"
    >
      <i
        className="fab fa-linkedin blink blink-blue"
        style={
          Object {
            "fontSize": 40,
          }
        }
      />
    </a>
  </div>
  <div
    className="animated-icon"
    style={
      Object {
        "margin": 8,
      }
    }
  >
    <a
      className=""
      href="https://www.facebook.com/profile.php?id=100010514613107"
      style={
        Object {
          "color": "black",
        }
      }
      target="_blank"
    >
      <i
        className="fab fa-facebook-square blink blink-blue"
        style={
          Object {
            "fontSize": 40,
          }
        }
      />
    </a>
  </div>
  <div
    className="animated-icon"
    style={
      Object {
        "margin": 8,
      }
    }
  >
    <a
      className=""
      href="https://www.instagram.com/harrywbell/"
      style={
        Object {
          "color": "black",
        }
      }
      target="_blank"
    >
      <i
        className="fab fa-instagram blink blink-purple"
        style={
          Object {
            "fontSize": 40,
          }
        }
      />
    </a>
  </div>
</div>
`;
